xarray/core/dataarray.py

```diff
@@ -6258,9 +6258,11 @@ def curvefit(
         Fit the exponential decay function to the data along the ``time`` dimension:
 
         >>> fit_result = da.curvefit("time", exp_decay)
-        >>> fit_result["curvefit_coefficients"].sel(param="time_constant")
+        >>> fit_result["curvefit_coefficients"].sel(
+        ...     param="time_constant"
+        ... )  # doctest: +NUMBER
```

Comment:
```suggestion
        >>> fit_result["curvefit_coefficients"].sel(
        ...     param="time_constant"
        ... ).round(3)
```
I think it's better to just round the values. You can do this in the other prints as well to make it more readable.
It's in general not a good idea to try to exactly print float values.

Reply:
Okay. I wanted to avoid `round()` because it might be confusing without further explanation of why it's there. It's not relevant to this example and not used in other examples AFAIK. All other examples print the full decimals.

Reply:
Is it confusing to display your results with decimals that matters? I don't think so, 

I'd find a cryptic comment like `# doctest: +NUMBER` more confusing and definitely have to google that because I didn't get the same results as the docstring.

Not sure what examples you're thinking of but I know that when I create examples I design the results to not require rounding (when possible) for exactlly these kind of flaky reasons.

Reply:
The `# doctest:` comments are not rendered, see here: https://xray--7893.org.readthedocs.build/en/7893/generated/xarray.DataArray.curvefit.html. It might be confusing for a developer looking at the docstring later, that I agree. But the `+NUMBER` thing is made exactly for this purpose.

By other examples I'm referring to other example usage in xarray methods, they always show the full decimal outputs. For example the example of creating a DataArray contains random numbers with full decimals: https://docs.xarray.dev/en/stable/generated/xarray.DataArray.html